File: guides/micronaut-microservices-distributed-tracing-zipkin-opentelemetry/bookrecommendation/java/src/main/java/example/micronaut/BookCatalogueClient.java
package example.micronaut;
/*
//tag::package[]
package example.micronaut;
//end::package[]
*/
//tag::imports[]

import io.micronaut.http.annotation.Get;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.retry.annotation.Recoverable;
import org.reactivestreams.Publisher;
//end::imports[]

//tag::harcoded[]
@Client("http://localhost:8081") // <1>
@Recoverable(api = BookCatalogueOperations.class)
//end::harcoded[]
//tag::clazz[]
public interface BookCatalogueClient extends BookCatalogueOperations {

    @Get("/books")
    Publisher<Book> findAll();
}
//end::clazz[]

File: guides/micronaut-microservices-distributed-tracing-jaeger/bookrecommendation/kotlin/src/main/kotlin/example/micronaut/BookCatalogueClient.kt
package example.micronaut
/*
//tag::package[]
package example.micronaut
//end::package[]
*/
//tag::imports[]

import io.micronaut.http.annotation.Get
import io.micronaut.http.client.annotation.Client
import io.micronaut.retry.annotation.Recoverable
import reactor.core.publisher.Flux
import org.reactivestreams.Publisher

//end::imports[]
//tag::harcoded[]
@Client("http://localhost:8081") // <1>
@Recoverable(api = BookCatalogueOperations::class)
//end::harcoded[]
//tag::clazz[]
interface BookCatalogueClient : BookCatalogueOperations {

    @Get("/books")
    override fun findAll(): Publisher<Book>
}
//end::clazz[]
